pandas/tests/arrays/sparse/test_array.py

```diff
@@ -478,3 +478,12 @@ def test_drop_duplicates_fill_value():
     result = df.drop_duplicates()
     expected = pd.DataFrame({i: SparseArray([0.0], fill_value=0) for i in range(5)})
     tm.assert_frame_equal(result, expected)
+
+
+def test_zero_sparse_column():
+    # GH 27781
+    df1 = pd.DataFrame({"A": SparseArray([0, 0, 0]), "B": [1, 2, 3]})
+    df2 = pd.DataFrame({"A": SparseArray([0, 1, 0]), "B": [1, 2, 3]})
+    result = df1.loc[df1["B"] != 2]
```

Comment:
Can you also compare this to a manually created dataframe i.e.

```
expected = pd.DataFrame({...})
tm.assert_frame_equal(result, expected)
```

Reply:
Added it, thanks for the suggestion.